### PR TITLE
Send IcM on notebook validation failure

### PR DESCRIPTION
**Pull request by kdestin**
# PR into Azure/azureml-examples

## Checklist

I have:

- [x] read and followed the contributing guidelines

## Changes

This pull-request adds an extra step to notebook validation that fires an IcM when notebook validation fails (see `sdk/readme.py`)

`readme.py` was run to regenerate relevant worklows.

### CODE REVIEW

File: .github/workflows/sdk-assets-component-component.yml [hunk 1]
```diff
@@ -51,3 +51,23 @@ jobs:
       with:
         name: component
         path: sdk/assets/component
+
+    - name: Send IcM on failure
+      if: ${{ failure() && (github.ref == 'refs/heads/main' || github.ref == 'ref/heads/release') }}
```
Comment (harneetvirk) [inline .github/workflows/sdk-assets-component-component.yml:56]: The release branch name will keep changing. You may need to add a contains 'release' filter

File: .github/workflows/sdk-assets-component-component.yml [hunk 2]
```diff
@@ -51,3 +51,23 @@ jobs:
...
+      uses: kdestin/generate-icm@v1
```
Comment (harneetvirk) [inline .github/workflows/sdk-assets-component-component.yml:57]: Dependency on the personal repo for any actions for validation is not recommended.

**Review by harneetvirk — APPROVED** [review]

**Review by paulshealy1 — APPROVED** [review]

**Review by harneetvirk — APPROVED** [review]

**Comment by harneetvirk** [comment]
Were you able to check why all the workflows are failing?

**Comment by kdestin** [comment]
> Were you able to check why all the workflows are failing?

Seems to be related to the fact that authentication fails when the source branch in from a fork

**Pull request closed without merging**